File: app/src/main/java/app/deadmc/devnetworktool/adapters/ParametersAdapter.java
package app.deadmc.devnetworktool.adapters;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.text.Spanned;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;

import app.deadmc.devnetworktool.R;

public class ParametersAdapter extends RecyclerView.Adapter<ParametersAdapter.ViewHolder> {

    ArrayList<Spanned> arrayList;
    Context context;

    public ParametersAdapter(Context context, ArrayList<Spanned> arrayList) {
        this.arrayList = arrayList;
        this.context = context;
    }

    public class ViewHolder extends RecyclerView.ViewHolder{
        private TextView text;
        public ViewHolder(View view) {
            super(view);
            text = (TextView)view.findViewById(R.id.textView);
        }
    }

    @Override
    public void onBindViewHolder(ParametersAdapter.ViewHolder viewHolder, int position) {
        final int currentPosition = position;
        viewHolder.text.setText(arrayList.get(position));
    }

    @Override
    public ParametersAdapter.ViewHolder onCreateViewHolder(ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.item_parameter, viewGroup, false);

        ViewHolder viewHolder = new ViewHolder(view);
        return viewHolder;
    }

    @Override
    public int getItemCount() {
        return arrayList.size();
    }

}

File: app/src/main/java/app/deadmc/devnetworktool/ui/adapters/ParametersAdapter.kt
package app.deadmc.devnetworktool.ui.adapters

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.text.Spanned
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import java.util.ArrayList

import app.deadmc.devnetworktool.R

class ParametersAdapter(internal var context: Context, internal var arrayList: ArrayList<Spanned>) : RecyclerView.Adapter<ParametersAdapter.ViewHolder>() {

    inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val text: TextView

        init {
            text = view.findViewById(R.id.textView)
        }
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, position: Int) {
        val currentPosition = position
        viewHolder.text.text = arrayList[position]
    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ViewHolder {
        val view = LayoutInflater.from(viewGroup.context).inflate(R.layout.item_parameter, viewGroup, false)

        return ViewHolder(view)
    }

    override fun getItemCount(): Int {
        return arrayList.size
    }

}
